app/helpers/search_helper.rb

```diff
@@ -66,5 +66,14 @@ def assign_material_link(material, action, extra={})
       link_to("Assign to a Class", action, extra.merge({:class=>"button"}))
     end
   end
-
+  
+  # return the grade_level_groups for which materials are authored
+  def authored_grade_level_groupes
```

Comment:
Yes I think this is the method.  My guess is actually the problem is in the `all_grade_levels` method. That method is loading in all of the Activities,Investigations, and ExternalActivities.  And then asking for the `grade_level_list` from that item.  The `grade_level_list` call will be second database request.  So this is inefficient because it is loading lots of stuff into memory and it is doing 1+N (roughly) database requests.

You should use Solr facets for this. 
https://github.com/sunspot/sunspot#faceting

Basically you'd construct a query that matches all of the Materials it should `facet` grade_levels and subject_areas.  Then you can ask for the 'rows' of grade_levels and 'rows' of subject_areas.  This row will have a value and count.  You can ignore the count, the values will be all of the grade levels.

In the case of grade_levels, it would also be OK to change the 'searchable' block of the Material to have a field of 'grade_level_group'.  We still want the database to store the individual grade levels, but it would be fine to have Solr store/index the grade_level_group.  If you did this, then you would no longer need to match individual grade_levels to grade_level_groups.